src/widgets/merch-catalog/ui/merch-filter/filter-controls/filter-controls.module.scss

```diff
@@ -0,0 +1,193 @@
+.filter-container {
+  display: flex;
+  flex-direction: column;
+  gap: 15px;
+}
+
+.filter-title-wrapper {
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+}
+
+.filter-title {
+  margin: 0;
+}
+
+.filter-search-wrapper {
+  display: flex;
+  gap: 5px;
+
+  padding: 5px;
+  border: 1px solid $color-gray-300;
+  border-radius: 5px;
+}
+
+.filter-search-input {
+  display: inline-flex;
+  width: 100%;
+  border: none;
+  outline: none;
+}
+
+.mobile-toggle-button {
+  .filter-toggle-arrow {
+    &.expanded {
+      transform: rotate(180deg);
+    }
+  }
+
+  @at-root .filter-container.mobile & {
+    cursor: pointer;
+
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+
+    width: 100%;
+    padding: 12px;
+    border: none;
+    border-radius: 5px;
+
+    font-size: 15px;
+    font-weight: 500;
+
+    background-color: transparent;
+    outline: none;
+
+    &.expanded {
+      background-color: $color-gray-50;
+    }
+  }
+}
+
+.filter-clear-button {
+  cursor: pointer;
+
+  padding: 4px 10px;
+  border: 1px solid $color-gray-300;
+  border-radius: 5px;
+
+  font-size: 12px;
+  color: $color-gray-600;
+
+  visibility: hidden;
+  opacity: 0;
+  background: none;
+  outline: none;
+
+  transition: color 0.2s ease;
+
+  &:hover {
+    border: 1px solid $color-black;
+    color: $color-black;
+  }
+
+  &.active {
+    visibility: visible;
+    opacity: 1;
+    transition-delay: 0s;
+  }
+}
+
+.mobile-clear-button {
+  cursor: pointer;
+
+  width: 100%;
+  margin-bottom: 15px;
+  padding: 8px;
+  border: 1px solid $color-gray-300;
+  border-radius: 5px;
+
+  font-size: 15px;
+  color: $color-gray-600;
+
+  background: none;
+  outline: none;
+
+  transition: color 0.2s ease;
+
+  &:hover {
+    color: $color-black;
+    background-color: $color-gray-50;
+  }
+}
+
+.filter-tags-wrapper {
+  display: flex;
+  flex-direction: column;
+  gap: 8px;
+}
+
+.filter-tag-label {
+  cursor: pointer;
+
+  display: flex;
+  gap: 10px;
+
+  width: 100%;
+  padding: 8px;
+  border: 1px solid transparent;
+  border-radius: 5px;
+
+  background-color: $color-gray-50;
+
+  &:hover {
+    background-color: $color-gray-200;
+  }
+}
+
+.filter-checkbox-custom {
+  position: relative;
+
+  display: inline-block;
+
+  width: 18px;
+  height: 18px;
+  border: 2px solid #ccc;
```

Comment:
Here and further on the code I see that the color is set directly, let's use css variables